codec/src/main/java/io/netty/handler/codec/MessageToMessageEncoder.java

```diff
@@ -99,9 +99,9 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
                 ctx.write(msg, promise);
             }
         } catch (EncoderException e) {
-            throw e;
+            promise.setFailure(e);
         } catch (Throwable t) {
-            throw new EncoderException(t);
+            promise.tryFailure(new EncoderException(t));
```

Comment:
Same as above, add a comment to explain the difference in using `try` vs `set`

Reply:
Actually I think using `setFailure` is better. Good catch